File: python/livre_magique.py
from dataclasses import dataclass

@dataclass
class LivreMagique:
    titre: str  
    magie_actuelle: int 

    @property
    def points_de_magie(self) -> int:
        return self.magie_actuelle

    def enchanter(self):
        self.magie_actuelle += 10

File: java/LivreMagique.java
/**
 * Représente un livre doté de propriétés mystiques.
 */
public class LivreMagique {
    private String titre;
    private int pointsDeMagie;

    /**
     * Constructeur pour initialiser le livre
     */
    public LivreMagique(String unTitre, int magieInitiale) {
        this.titre = unTitre;
        this.pointsDeMagie = magieInitiale;
    }
    
    public String getTitre() {
        return titre;
    }

    public int getPointsDeMagie() {
        return pointsDeMagie;
    }

    /**
     * Méthode qui manipule les attributs 
     * Cette méthode "enchanter" augmente la puissance du livre.
     */
    public void enchanter() {
        this.pointsDeMagie = this.pointsDeMagie + 10;
    }
}